### PR TITLE
Rewrite Cardinality(SUBSET S) into 2^Cardinality(S)

### LINKED ISSUE #1360: [FEATURE] Cardinality of a powerset fails
The following test fails:

```tla
Set123567 ==
    LET one == 1 IN
    LET two == 2 IN
    LET three == one + two IN
    \* 3 and three are equal but the set constructor will miss that
    { one, two, 3, three, three, 5, 6, 7, 7 }

\* The set { 2, 6 }
\* that has ghost cells 1, 3, three, 5 in its internal representation
Set26 ==
    { i \in Set123567: i % 2 = 0 }

\* The set { 2, 3, 6 }
\* that has ghost cells 1, 3, three, 5 and non-ghost cells 2, 3, 6
Set263 ==
    Set26 \union { 3 }

TestPowersetCardinality ==
    Cardinality(SUBSET Set263) = 8
```

When checking `TestPowersetCardinality` as an invariant, a counterexample is reported.

### PR DESCRIPTION
**Pull request by konnov**
Closes #1360. Simply preprocess `Cardinality(SUBSET S)`.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

**Comment by konnov** [comment]
> ...he/io/annotations/parser/CommentPreprocessor.scala | 92.77% <0.00%> (-1.21%)

CodeCov went a bit wild and calculated code coverage for the file that has not been touched by the PR. There is nothing to worry about, just keep in mind that it can be imprecise.

File: tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ExprOptimizer.scala [hunk 1]
```diff
@@ -80,6 +80,12 @@ class ExprOptimizer(nameGen: UniqueNameGenerator, tracker: TransformationTracker
       val bMinusA = OperEx(TlaArithOper.minus, right, left)(intTag)
       OperEx(TlaArithOper.plus, bMinusA, ValEx(TlaInt(1))(intTag))(intTag)
 
+    case OperEx(TlaFiniteSetOper.cardinality, OperEx(TlaSetOper.powerset, set)) =>
+      // A pattern that emerged in issue #1360
+      // Cardinality(SUBSET S) is equivalent to 2^Cardinality(S).
+      val cardEx = OperEx(TlaFiniteSetOper.cardinality, set)(set.typeTag)
```
Comment (Kukovec) [inline tla-pp/src/main/scala/at/forsyte/apalache/tla/pp/ExprOptimizer.scala:86]:
```suggestion
      val cardEx = OperEx(TlaFiniteSetOper.cardinality, set)(intTag)
```

**Review by Kukovec — COMMENTED** [review]
Wrong tag?

**Review by Kukovec — APPROVED** [review]
LGTM

**Pull request merged**